### DISCUSSION TITLE
Async init data in x-init

### DISCUSSION THREAD

**Discussion by alexej-d** [alpinejs/alpine#1538] [1. Help]
Hey everyone, consider the following code:

```
<div x-data="loader" x-init="load('module', { data: { items: ['a', 'b', 'c'] } })">
<template x-for="(item, index) in items" :key="index">
...
</template>
</div>
```

and the following data directive:

```
Alpine.data('loader', () => {
  let promises = {}
  return {
    async load(type, info = {}) {
      const { data, ...rest } = info

      //apply default data
      Object.keys(data).map((key) => (this[key] = data[key]))

      if (!promises[type])
          promises[type] = await import( `modules/${type}`)

      promises[type].default(rest, this)
    },
  }
})
```

Applying the default data with Objekt.keys before the await works, but doing so after gives me en error `ReferenceError: items is not defined`. Is this expected or is this a bug?

**Comment by SimoTod** [comment]
It's expected, you don't have any properties until the async callback has finished so Alpine will error at the first rendering.
